WordPress/gutenberg
# [RNMobile] RichText should delete its Enter in Heading block

## Description
Fix for https://github.com/wordpress-mobile/gutenberg-mobile/issues/4322

### Related PRs
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/4365

## How has this been tested?

### Test 1
1. Using the demo app on Android 12
2. Insert a heading block and input some words
3. Place the cursor in the middle of a word and hit the Enter key on the virtual keyboard
4. Notice the block split into 2 heading blocks and the caret being at the beginning of the second block
5. Hit the Enter key on the virtual keyboard again
6. Notice that a new paragraph block has been added between the two heading blocks, **and that the second heading block doesn't have an empty line in its start**

### Test 2

Follow the same steps as in Test 1, but on an Android 11 device/emulator. This is a test to confirm that the behavior is not broken on current Android version.

## Types of changes
1. Have the heading block pass the `deleteEnter` prop to its RichText when on native mobile. I opted for a platform conditional (versus for example forking the edit.js into edit.native.js) to avoid forking the block for this kind of small tweak. Also, opted for [not defaulting the RichText component](https://github.com/WordPress/gutenberg/blob/1b15cf17199126b73aeb3644be7527e7806e22fc/packages/block-editor/src/components/rich-text/index.native.js#L97) to `deleteEnter=true` since that would increase the impact of the change to too many other components/blocks. The fix in this PR is targeting the heading block only. If the same issue exists in other blocks, we can reconsider where to have the fix.

## Checklist:
